crates/primitives/src/signature/sig.rs

```diff
@@ -271,15 +251,7 @@ impl Signature<()> {
         s: U256,
         parity: T,
     ) -> Result<Self, SignatureError> {
-        Ok(Self { inner: (), v: parity.try_into().map_err(Into::into)?, r, s })
-    }
-}
-
-impl<S: Copy> Signature<S> {
-    /// Returns the inner ECDSA signature.
-    #[inline]
-    pub const fn into_inner(self) -> S {
```

Comment:
we can keep this as deprecated for cfg k256 and panic on error to avoid breaking change